### PR TITLE
Softsign op

### PR DESCRIPTION
**Pull request by nswamy**
## Description ##
Add [Softsign Activation function](https://sefiks.com/2017/11/10/softsign-as-a-neural-networks-activation-function/) 

### Essentials ###
- [ ] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by nswamy** [comment]
@cjolivier01 @anirudh2290

**Comment by cjolivier01** [comment]
does check_numeric_gradient() pass?

**Comment by nswamy** [comment]
yes it does pass the numeric_gradient check

File: tests/python/unittest/test_operator.py [hunk 1]
```diff
@@ -486,6 +486,20 @@ def fsigmoid(a):
     check_symbolic_forward(y, [xa], [ya])
     check_symbolic_backward(y, [xa], [np.ones(shape)], [ya * (1 - ya)])
 
+def test_softsign():
```
Comment (anirudh2290) [inline tests/python/unittest/test_operator.py]: add @with_seed()

File: src/operator/tensor/elemwise_unary_op_basic.cc [hunk 2]
```diff
@@ -106,6 +106,23 @@ The storage type of ``sigmoid`` output is always dense
 
 MXNET_OPERATOR_REGISTER_BINARY_WITH_SPARSE_CPU(_backward_sigmoid,
                                                unary_bwd<mshadow_op::sigmoid_grad>);
+// softsign
+MXNET_OPERATOR_REGISTER_UNARY(softsign)
+MXNET_ADD_SPARSE_OP_ALIAS(softsign)
```
Comment (eric-haibin-lin) [inline src/operator/tensor/elemwise_unary_op_basic.cc]: This operator doesn't have a sparse implementation. Sparse op alias should not be registered.

File: src/operator/nn/activation-inl.h [hunk 3]
```diff
@@ -168,6 +173,10 @@ void ActivationGradComputeImpl(const ActivationParam &param, const OpContext &ct
         ActivationBackward<xpu, mshadow_op::softrelu, mshadow_op::softrelu_grad, DType>(
             ctx, out_grad, out_data, req, output);
         break;
+      case activation::kSoftSign:
```
Comment (eric-haibin-lin) [inline src/operator/nn/activation-inl.h]: Are you sure this is correct? 

`ActivationGradComputeImpl` takes `out_grad` and `out_data` to calculate `output`. In another word, 
for `y = activation(x)`, it calculates `dx = _backward_activation(dy, y)`, not `dx = _backward_activation(dy, x)`.

File: tests/python/unittest/test_operator.py [hunk 4]
```diff
@@ -486,6 +486,21 @@ def fsigmoid(a):
     check_symbolic_forward(y, [xa], [ya])
     check_symbolic_backward(y, [xa], [np.ones(shape)], [ya * (1 - ya)])
 
+@with_seed()
+def test_softsign():
```
Comment (eric-haibin-lin) [inline tests/python/unittest/test_operator.py]: Is there unit test for `Activation(act_type='softsign')`???

**Pull request merged**